src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -600,14 +629,67 @@ public override bool HasDynamicDependencies
                 // This node's EETypeOptionalFields node may change if this EEType implements interfaces
                 // that are used since the dispatch map table index is computed once we know the interface
                 // layout later on in compilation.
-                return _type.RuntimeInterfaces.Length > 0 && _constructed;
+
+                // Also, if any of the implemented interfaces have variance, variant dispatch might result in
+                // slots being necessary.
+                foreach (var intfType in _type.GetClosestMetadataType().RuntimeInterfaces)
```

Comment:
I would hope that we will not keep the "if the type has any interfaces, it has dynamic dependencies" logic here for too long.

When we get rid of it, we would still need "if the type has any *variant* interfaces, it has dynamic dependencies".